### DISCUSSION TITLE
Exceptions X Domain Notification

### DISCUSSION THREAD

**Discussion by Jvprotano** [balta-io/forum#692] [Tira Dúvidas] [answered]
Estou fazenda a carreira "Advanced .NET Developer", neste momento estou no curso "Aplicando Orientação a Objetos em Projetos Reais com C# 11 e .NET 7".

A minha dúvida é algo que não ficou claro para mim na verdade, relacionado ao uso de Domain Notification, como o Flunt, e Exceptions. Existem métricas que mostrem a vantagem de um sobre o outro? No cenário de adotar para um grande projeto, existe uma "melhor escolha" nesse sentido? 
Pergunto, pois em um dos cursos foi demonstrado o uso do Flunt, que foi uma abordagem nova para mim e que gostei bastante, mas no curso que estou nesse momento foram adotadas as exceptions, que é a abordagem em que estou mais habituado. 

Peço desculpas se o tópico não for para esse tipo de dúvida. Abraço!

**Comment by henriqueweege** [comment, marked as answer]
Oi, @Jvprotano! Tudo joia?

Usar exceções para controle de fluxo é considerado um anti-pattern. Caso queira, você pode ler mais sobre isso [nesse artigo](https://enterprisecraftsmanship.com/posts/exceptions-for-flow-control/).
Além disso, existem custos de processamento e memória que estão atrelados as exceções. Portanto, por existirem maneiras mais eficientes de fazer essas tratativas, exceções costumam ser evitadas.

Valeeeu!
